context/datapath.go

```diff
@@ -534,37 +476,19 @@ func (dpNode *DataPathNode) ActivateUpLinkPdr(smContext *SMContext, defQER *QER,
 
 	curULTunnel := dpNode.UpLinkTunnel
 	for name, ULPDR := range curULTunnel.PDR {
-		ULDestUPF := curULTunnel.DestEndPoint.UPF
 		ULPDR.QER = append(ULPDR.QER, defQER)
 
 		// Set Default precedence
 		if ULPDR.Precedence == 0 {
 			ULPDR.Precedence = defPrecedence
 		}
 
-		var iface *UPFInterfaceInfo
-		if dpNode.IsANUPF() {
-			iface = ULDestUPF.GetInterface(models.UpInterfaceType_N3, smContext.Dnn)
-		} else {
-			iface = ULDestUPF.GetInterface(models.UpInterfaceType_N9, smContext.Dnn)
+		ULPDR.PDI.SourceInterface = SourceInterface{InterfaceValue: SourceInterfaceAccess}
+		ULPDR.PDI.LocalFTeid = &FTEID{
+			Ch: true,
 		}
-
-		if upIP, err := iface.IP(smContext.SelectedPDUSessionType); err != nil {
```

Comment:
I think we are removing code which supports iUPF

Reply:
What's iUPF? 

The code you are referring to is not quite deleted but moved, please look at line 520

Reply:
@gruyaume, FYI, iUPF is the intermediate UPF (N3 and N9 interfaces)